### PR TITLE
native: fix warning error OS X

### PR DESCRIPTION
**Pull request by thomaseichinger**
rebased upon ~~#3290~~ and #3390 to build again after merge of #3319.
@emmanuelsearch can you verify?

### CODE REVIEW

**Comment by gebart** [comment]
@thomaseichinger could you add comments to the `#else` and `#endif` to the places where they are more than only 2-3 lines apart?

**Comment by emmanuelsearch** [comment]
Works for me!

**Comment by gebart** [comment]
@thomaseichinger please squash

**Comment by thomaseichinger** [comment]
Updated after merge of #3390 and squashed.

**Comment by thomaseichinger** [comment]
@emmanuelsearch ping

**Comment by emmanuelsearch** [comment]
No warning for me either when I build with 10.10.5. However, if I run the Default example I get the following runtime error:
```
kernel_init(): This is RIOT! (Version: 2014.12-2274-g066fe-pr-3393)
kernel_init(): jumping into first task...
UART0 thread started.
uart0_init() [OK]
Native LTC4150 initialized.
No hwtimer left.
/Users/emmanuelB/Documents/work/CODING/RIOT/RIOT/examples/default/bin/native/default.elf: ltc4150_enable_int: hwtimer_set
make: *** [term] Error 1
```

**Comment by kaspar030** [comment]
@emmanuelsearch Hmm, ltc4150 has been kicked from native a while ago. 

@thomaseichinger Maybe this PR just needs a rebase?

**Comment by thomaseichinger** [comment]
rebased. @emmanuelsearch could you try to run default again?

**Comment by thomaseichinger** [comment]
OK, rebasing introduced a new problem. You should see something like: 
```bash
ld: file too small (length=8) file '/Users/thomas/code/RIOT/examples/gnrc_networking/bin/native/gnrc.a' for architecture i386
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by emmanuelsearch** [comment]
actually it runs like a charm on my machine (OS X 10.10.5)

**Comment by emmanuelsearch** [comment]
but when I try to compile gnrc_networking I get the following errors:

```
/Users/emmanuelB/Documents/work/CODING/RIOT/RIOT/sys/shell/commands/sc_netif.c:701:16: error: 
      format specifies type 'unsigned short' but the argument has type 'int'
      [-Werror,-Wformat]
               IPV6_MIN_MTU, UINT16_MAX);
               ^~~~~~~~~~~~
/Users/emmanuelB/Documents/work/CODING/RIOT/RIOT/sys/include/net/ipv6.h:42:25: note: 
      expanded from macro 'IPV6_MIN_MTU'
#define IPV6_MIN_MTU    (1280)
                        ^~~~~~
/Users/emmanuelB/Documents/work/CODING/RIOT/RIOT/sys/shell/commands/sc_netif.c:701:30: error: 
      format specifies type 'unsigned short' but the argument has type 'int'
      [-Werror,-Wformat]
               IPV6_MIN_MTU, UINT16_MAX);
                             ^~~~~~~~~~
/usr/include/stdint.h:85:27: note: expanded from macro 'UINT16_MAX'
#define UINT16_MAX        65535
                          ^~~~~
2 errors generated.
make[3]: *** [/Users/emmanuelB/Documents/work/CODING/RIOT/RIOT/examples/gnrc_networking/bin/native/shell_commands/sc_netif.o] Error 1
make[2]: *** [ALL--/Users/emmanuelB/Documents/work/CODING/RIOT/RIOT/sys/shell/commands] Error 2
make[1]: *** [ALL--/Users/emmanuelB/Documents/work/CODING/RIOT/RIOT/sys] Error 2
make: *** [all] Error 2
```

**Comment by thomaseichinger** [comment]
I fixed these. @emmanuelsearch could you try again?

**Comment by emmanuelsearch** [comment]
Fix works on my side too. I get the "ld: file too small" error now ;)

**Comment by thomaseichinger** [comment]
The problem is that an empty archive `gnrc.a` gets created but there are never files added to it. Executing the linker command manually without the mentioned archive succeeds and results in a working binary.

**Comment by thomaseichinger** [comment]
@emmanuelsearch I added a workaround for the empty archive problem. Could you test again?

**Comment by emmanuelsearch** [comment]
great. I can now run gnrc_networking. Have not tested more than the shell though. As discussed offline, I suspect we still have tap setup issues, right?

**Comment by emmanuelsearch** [comment]
Assuming tap setup issues are a separate issue, I'd ACK this ;)

**Comment by kaspar030** [comment]
please rebase... I'm testing this now.

**Comment by thomaseichinger** [comment]
rebased

**Comment by emmanuelsearch** [comment]
I tried it on OS X 10.10.5. It compiles without warning and runs fine: I get a functioning shell with default and gnrc_networking.

**Comment by emmanuelsearch** [comment]
ACK.

**Comment by emmanuelsearch** [comment]
(once Travis is green ;)

File: Makefile.include [hunk 1]
```diff
@@ -213,7 +216,7 @@ all: ..compiler-check ..build-message $(USEPKG:%=${BINDIR}%.a) $(APPDEPS)
 	$(AD)DIRS="$(DIRS)" "$(MAKE)" -C $(CURDIR) -f $(RIOTBASE)/Makefile.application
 ifeq (,$(RIOTNOLINK))
 ifeq ($(BUILDOSXNATIVE),1)
-	$(AD)$(if $(CPPMIX),$(CXX),$(LINK)) $(UNDEF) -o $(ELFFILE) $(BASELIBS) $(LINKFLAGS) $(LINKFLAGPREFIX)-no_pie
+	$(AD)$(if $(CPPMIX),$(CXX),$(LINK)) $(UNDEF) -o $(ELFFILE) $(filter-out $(EMPTYARCHIVES), $(BASELIBS)) $(LINKFLAGS) $(LINKFLAGPREFIX)-no_pie
```
Comment (Kijewski) [inline Makefile.include]: Please try
```make
$(shell find $(BASELIBS) -size +8c)
```
which should return the .a files that are bigger than eight bytes. The .a header is 8 bytes.

Comment (thomaseichinger) [inline Makefile.include]: This leaves me with a linker command without any archives when I call it with `make clean all`. The archives cannot be found by `find` as they are removed before. Issuing a consecutive `make all` succeeds then.

Comment (Kijewski) [inline Makefile.include]: I right, `$$(find $(BASELIBS) -size +8c)` then? :)

Comment (thomaseichinger) [inline Makefile.include]: This works! Thanks a lot! Updated the commit.

**Comment by thomaseichinger** [comment]
Travis passed. Go

**Pull request merged**